src/main/java/project/myblog/service/PostService.java

```diff
@@ -0,0 +1,64 @@
+package project.myblog.service;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import project.myblog.domain.Member;
+import project.myblog.domain.Post;
+import project.myblog.exception.BusinessException;
+import project.myblog.exception.ExceptionCode;
+import project.myblog.repository.PostRepository;
+import project.myblog.web.dto.post.PostRequest;
+import project.myblog.web.dto.post.PostResponse;
+
+import java.util.Optional;
+
+@Transactional
+@Service
+public class PostService {
+    private final PostRepository postRepository;
+    private final MemberService memberService;
+
+    public PostService(PostRepository postRepository, MemberService memberService) {
+        this.postRepository = postRepository;
+        this.memberService = memberService;
+    }
+
+    public Long createPost(String email, PostRequest requestDto) {
+        Member member = memberService.findMemberByEmail(email);
+        return postRepository.save(requestDto.toEntity(member)).getId();
+    }
+
+    public PostResponse findPost(Long postId) {
+        Post post = findPostById(postId);
+        return new PostResponse(post);
+    }
+
+    public Long updatePost(String email, Long postId, PostRequest requestDto) {
+        Member member = memberService.findMemberByEmail(email);
+
+        Optional<Post> post = postRepository.findById(postId);
+        if (post.isPresent()) {
+            if (post.get().isAuthorization(member)) {
+                return post.get().update(requestDto.getTitle(), requestDto.getContents());
+            }
+            throw new BusinessException(ExceptionCode.POST_AUTHORIZATION);
+        }
+        return postRepository.save(requestDto.toEntity(member)).getId();
+    }
+
+    public void deletePost(String email, Long postId) {
+        Member member = memberService.findMemberByEmail(email);
+        Post post = findPostById(postId);
+
+        if (post.isAuthorization(member)) {
+            postRepository.delete(post);
+        } else {
+            throw new BusinessException(ExceptionCode.POST_AUTHORIZATION);
+        }
```

Comment:
```suggestion
        post.delete(member);
```
삭제여부를 `deleted`로 따로 관리한다면 서비스 로직을 이렇게 작성하고, `isAutorization`메서드는 delete메서드 내부로 가져갈 수 있을 것 같아요.

관리하지 않을 거라면 이렇게 해보는건 어떨까요??
```suggestion
        post.checkOwner(member);
        postRepository.delete(post);
```

Reply:
일단 관리하지 않는 방향으로 할게요!!
고민 해보고 정해요!